engine-kommand/src/commonMain/kotlin/com/saveourtool/processbuilder/KommandEngine.kt

```diff
@@ -0,0 +1,66 @@
+package com.saveourtool.processbuilder
+
+import com.saveourtool.processbuilder.utils.CurrentOs
+import com.saveourtool.processbuilder.utils.cwd
+import com.saveourtool.processbuilder.utils.getCurrentOs
+import com.saveourtool.processbuilder.utils.logger
+import com.saveourtool.processbuilder.utils.stderr
+import com.saveourtool.processbuilder.utils.stdout
+
+import com.kgit2.process.Command
+
+import kotlinx.coroutines.CoroutineScope
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.async
+
+private typealias CommandWithArgs = Pair<String, Array<String>>
+
+object KommandEngine : Engine {
+    private val scope = CoroutineScope(Dispatchers.Default)
+
+    override suspend fun execute(
+        command: String,
+        config: ProcessBuilderConfig,
+    ): ExecutionResult {
+        val (cmd, args) = getCommandWithArgs(escapeDoubleTicks(command)).also {
+            logger.info { "${it.first} ${it.second.joinToString(" ")}" }
+        }
+        val stdout = config.stdout
+        val stderr = config.stderr
+        val child = Command(cmd)
+            .args(*args)
+            .cwd(config.workingDirectory)
+            .stdout(stdout)
+            .stderr(stderr)
+            .spawn()
+        return scope.async {
```

Comment:
If you only have a **single** asynchronous result, `withContext()` should be used instead of `async()`/`await()`.

Yet, this code should be rewritten with **two** or **three** `async()`/`await()` pairs, because you're waiting for:

 1. the process to complete (i.e. the exit code)
 2. the `stdout` to be fully read
 3. the `stderr` to be fully read

`stdout` and `stderr` should be read in two independent threads, so on the JVM you need a separate coroutine context backed by a pool (and `ExecutorService`) of two threads to avoid possible thread starvation.

I believe we've already discussed this earlier, let's re-visit the problem once again (offline).

Reply:
Will update `kotlinx.coroutines` to `1.7.0` and use `Dispatchers.IO` for both `jvm` and `native` implementations.

Reply:
Will run stdout/stderr reader jobs on `Dispatchers.IO` while the parent job will be run on current suspend function context.

Reply:
Now using `Channel`s over `Dispatchers.IO` in order to read stdout and stderr